model/FullText.inc.php

```diff
@@ -59,52 +59,33 @@ public static function indexItem(Zotero_Item $item, $data) {
 	}
 	
 	private static function indexItemInElasticsearch($libraryID, $key, $version, $timestamp, $data) {
-		$type = self::getWriteType();
-		
-		$id = $libraryID . "/" . $key;
-		$doc = [
-			'id' => $id,
-			'libraryID' => $libraryID,
-			'content' => (string) $data->content,
-			// We don't seem to be able to search on _version, so we duplicate it here
+		$params = [
+			'index' => self::$elasticsearchType . "_index_write",
+			'type' => self::$elasticsearchType,
+			'routing' => $libraryID,
+			'id' => $libraryID . "/" . $key,
+			'version_type' => 'external_gte', // Greater or equal
 			'version' => $version,
-			// Add "T" between date and time for Elasticsearch
-			'timestamp' => str_replace(" ", "T", $timestamp)
+			'body' => [
+				'libraryID' => $libraryID,
+				'content' => (string) $data->content,
+				// We don't seem to be able to search on _version, so we duplicate it here
+				'version' => $version,
+				// Add "T" between date and time for Elasticsearch
+				'timestamp' => str_replace(" ", "T", $timestamp)
+			]
 		];
+
 		foreach (self::$metadata as $prop) {
 			if (isset($data->$prop)) {
-				$doc[$prop] = (int) $data->$prop;
+				$params['body'][$prop] = (int) $data->$prop;
 			}
 		}
+
 		$start = microtime(true);
-		$doc = new \Elastica\Document($id, $doc, self::$elasticsearchType);
-		$doc->setVersion($version);
-		$doc->setVersionType('external');
-		$doc->setRouting($libraryID);
-		try {
-			$response = $type->addDocument($doc);
-		}
-		catch (Exception $e) {
-			$msg = $e->getMessage();
-			if (preg_match('/version conflict, current \[([0-9]+)\], provided \[([0-9]+)\]/', $msg, $matches)) {
-				if ($matches[1] == $matches[2]) {
-					error_log("WARNING: " . $msg);
-					return;
-				}
-			}
-			throw $e;
-		}
+		// Throws exception if version is lower. Overwrites if greater or equal
+		Z_Core::$ES->index($params);
 		StatsD::timing("elasticsearch.client.item_fulltext.add", (microtime(true) - $start) * 1000);
-		if ($response->hasError()) {
-			$msg = $response->getError();
-			if (preg_match('/version conflict, current \[([0-9]+)\], provided \[([0-9]+)\]/', $msg, $matches)) {
```

Comment:
Changed version_type from external to external_gte to avoid this check. Not sure if this decision is right.